test/unit/ngsiv2/plugins/bidirectional-plugin_test.js

```diff
@@ -0,0 +1,416 @@
+        iotAgentLib.clearAll(function() {
+            iotAgentLib.deactivate(done);
+        });
+    });
+
+    describe('When a new provisioning request arrives to the IoTA with bidirectionality', function() {
+        beforeEach(function() {
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/subscriptions', utils.readExampleFile(
+                './test/unit/ngsiv2/examples/subscriptionRequests/bidirectionalSubscriptionRequest.json'))
+                .reply(201, null, {'Location': '/v2/subscriptions/51c0ac9ed714fb3b37d7d5a8'});
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v1/updateContext', utils.readExampleFile(
+                    './test/unit/examples/contextRequests/createBidirectionalDevice.json'))
+                .reply(200, utils.readExampleFile(
+                    './test/unit/examples/contextResponses/createBidirectionalDeviceSuccess.json'));
+
+        });
+
+        it('should subscribe to the modification of the combined attribute with all the variables', function(done) {
+            request(options, function(error, response, body) {
+                should.not.exist(error);
+                contextBrokerMock.done();
+                done();
+            });
+        });
+    });
+
+    describe('When a device with bidirectionality subscriptions is removed', function() {
+        var deleteRequest = {
+            url: 'http://localhost:' + iotAgentConfig.server.port + '/iot/devices/Light1',
+            method: 'DELETE',
+            headers: {
+                'fiware-service': 'smartGondor',
+                'fiware-servicepath': '/gardens'
+            }
+        };
+
+        beforeEach(function() {
+            contextBrokerMock = nock('http://192.168.1.1:1026')
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v2/subscriptions', utils.readExampleFile(
+                    './test/unit/ngsiv2/examples/subscriptionRequests/bidirectionalSubscriptionRequest.json'))
+                .reply(201, null, {'Location': '/v2/subscriptions/51c0ac9ed714fb3b37d7d5a8'});
+
+            contextBrokerMock
+                .matchHeader('fiware-service', 'smartGondor')
+                .matchHeader('fiware-servicepath', '/gardens')
+                .post('/v1/updateContext', utils.readExampleFile(
+                    './test/unit/examples/contextRequests/createBidirectionalDevice.json'))
+                .reply(200, utils.readExampleFile(
+                    './test/unit/examples/contextResponses/createBidirectionalDeviceSuccess.json'));
+
+            contextBrokerMock = nock('http://192.168.1.1:1026')
```

Comment:
Same effect I have observed in the other test file... it seems that this assignment for the `contextBrokerMock` variable overrides the previous one.

Reply:
Answered in a previous comment

Reply:
NTC (to close this comment thread... discussion has been diverted to the "previous comment")